enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,416 @@
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type NodeDeviceMapStatus struct {
+	Phase string `json:"phase,omitempty"`
+	// Blockdevices will be containing the output of following command unmarshalled into the structured
+	// "lsblk -o name,model,vendor,kname,pkname,type,uuid,tran,ro,rm,parttype,serial,size,rota,mountpoint"
+	Blockdevices []Device `json:"blockdevices"`
+}
+
+type Device struct {
+	Name       string   `json:"name,omitempty"`
+	Model      string   `json:"model,omitempty"`
+	Vendor     string   `json:"vendor,omitempty"`
+	Kname      string   `json:"kname,omitempty"`
+	Pkname     string   `json:"pkname,omitempty"`
+	Type       string   `json:"type,omitempty"`
+	UUID       string   `json:"uuid,omitempty"`
+	Tran       string   `json:"tran,omitempty"`
+	Ro         string   `json:"ro,omitempty"`
+	Parttype   string   `json:"parttype,omitempty"`
+	Serial     string   `json:"serial,omitempty"`
+	Size       string   `json:"size,omitempty"`
+	Mountpoint string   `json:"mountpoint,omitempty"`
+	Rm         string   `json:"rm,omitempty"`
+	Children   []Device `json:children,omitempty`
+}
+```
+
+#### Workflow of Device Discovery
+- The discovery of local devices will be done by the `LocalVolumeDiscovery` CR. Once the `LocalVolumeDiscovery` CR is deployed, the controller of the `LocalVolumeDiscovery` CR will be managing a `NodeDeviceMap` CR for each node that matched the `nodeSelector`. The `NodeDeviceMap` CR will be exposing the available local devices of the node in its status. These information will mostly be the output of  `lsblk -o name,model,vendor,kname,pkname,type,uuid,tran,ro,rm,parttype,serial,size,rota,mountpoint`
+
+Example of `LocalVolumeDiscovery` CR:
+```yaml
+apiVersion: local.storage.openshift.io/v1alpha1
+kind: LocalVolumeDiscovery
+metadata:
+  name: example-autodetect
+  namespace: localStorage
+spec:
+  nodeSelector:
+    nodeSelectorTerms:
+      - matchExpressions:
+          - key: kubernetes.io/hostname
+            operator: In
+            values:
+              - worker-0
+              - worker-1
+status:
+  phase: Discovered
+```
+
+Example of `NodeDeviceMap` CR after detection:
+```yaml
+apiVersion: local.storage.openshift.io/v1alpha1
+kind: NodeDeviceMap
+metadata:
+  name: example-volume-detect-1
+  namespace: localStorage
+spec:
+  nodeName: worker-0
+  tolerations: null
+Status:
+  Phase: Discovered
+  Blockdevices:
+    - name: sda
+      model: SAMSUNG MZ7LN512
+      vendor: 'ATA     '
+      kname: sda
+      pkname: null
+      type: disk
+      uuid: null
+      tran: sata
+      ro: '0'
+      parttype: null
+      serial: S1ZKNXAH103764
+      size: 477G
+      rota: '0'
+      mountpoint: null
+```
+Now, once we have the detected devices the administrator can create the localPVs by explicitly selecting the disks. This can be done by the `localVolume` CR. The other option would be to create localPVs via the AutoDetectVolume CR and passing the inclusion filters in it.
```

Comment:
This section from lines 305-315 is what I was thinking would be earlier in the document before the design details. Understanding the high level flow before all the details is helpful.

Reply:
done